File: May-18/07-05-18/python/01-Marut-and-strings.py

def solu(data, n):
    cnt1 = 0
    cnt2 = 0
    v1 = 0
    v2 = 0
    for i in data:
        v1 = ord(i) - ord('a')
        v2 = ord(i) - ord('A')
        if(v1 >= 0 and v1 <= 25):
            cnt1 += 1
        
        if(v2 >= 0 and v2 <= 25):
            cnt2 += 1
        
    if( (cnt1 == 0 and cnt2 == 0) or (n < 1 or n > 100) ):
        print("Invalid Input")
    else:
        print(min(cnt1, cnt2))


data = input()
t = 0
try:
    t = int(data)
except:
    t = -1

if(t < 1 or t > 10):
    print("Invalid Test")
else:
    for _ in range(0, t):
        data = input()
        n = len(data)
        solu(data, n)

File: May-18/07-05-18/java/01-Marut-and-strings.java

import java.util.*;
import java.io.*;


class TestClass {
    static int v1, v2, cnt1, cnt2;
    static void solu(String str, int n)
    {
        v1 = v2 = cnt1 = cnt2 = 0;
        for(int i = 0; i < n; i++)
        {
            v1 = str.charAt(i) - 'a';
            v2 = str.charAt(i) - 'A';
            if(v1 >= 0 && v1 <= 25)
                cnt1++;
            else if(v2 >= 0 && v2 <= 25)
                cnt2++;
        }
        if(( cnt1 == 0 && cnt2 == 0) || (n < 1 || n > 100))
        {
            System.out.println("Invalid Input");
            return;
        }
        System.out.println(Math.min(cnt1, cnt2));
        return;
    }
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        String data = input.readLine();
        int t = 0;
        try
        {    t = Integer.parseInt(data);
        }
        catch(Exception e)
        {
            t = -1;
        }
        if(t < 1 || t > 10)
            System.out.println("Invalid Test");
        else
        {
            while(t-- > 0)
            {
                String str = input.readLine();
                int n = str.length();
                solu(str, n);
            }
        }
        
    }
}
